src/components/dataset/query/QueryView.jsx

```diff
@@ -78,6 +80,10 @@ export class QueryView extends React.PureComponent {
     if (datasetPolicies == null || dataset.id !== datasetId) {
       dispatch(getDatasetPolicy(datasetId));
     }
+
+    if (userRole == null || dataset.id !== datasetId) {
```

Comment:
I think this if statement and the one above it are race conditions (you'll NPE if dataset is null...eg hasn't loaded yet.  You can do this to protect

```suggestion
    if (userRole == null || dataset?.id !== datasetId) {
```

Reply:
We specifically set the `dataset` object in state to default to an empty object. See https://github.com/DataBiosphere/jade-data-repo-ui/pull/1065/files#diff-514b4d730e0e38947ad19c45203aace96bc11ac05259adf1a519c832f3137e85R8

Reply:
oooooh ok.  Then not sure why we're checking for null on line 76